torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -521,6 +522,26 @@ def speculate_branch(branch):
         if not same_treespec.as_python_constant():
             unimplemented("Expected branches to return the same pytree structure.")
 
+        def diff_meta(tensor_vars1, tensor_vars2):
+            assert all(
+                isinstance(var, TensorVariable) for var in tensor_vars1 + tensor_vars2
+            )
+            all_diffs = []
+            for i, (var1, var2) in enumerate(zip(tensor_vars1, tensor_vars2)):
+                # We check the meta data associated with meta["example_value"]
+                meta1 = _extract_tensor_metadata(var1.proxy.node.meta["example_value"])
+                meta2 = _extract_tensor_metadata(var2.proxy.node.meta["example_value"])
+                if meta1 != meta2:
+                    all_diffs.append((f"pair{i}:", meta1, meta2))
```

Comment:
Does this also do strides? This might make cond difficult to use, but we can cross that bridge when it happens (e.g. if strides mismatch we can always just return .contiguous())

Reply:
It includes strides. Sounds good!